JS.md

```diff
@@ -243,8 +242,30 @@ of `exports`. Let `moduleNamespace` be the result of calling
 [`ModuleNamespaceCreate(moduleRecord, exportStrings)`](http://tc39.github.io/ecma262/#sec-modulenamespacecreate).
 Set `moduleRecord.[[Namespace]]` to `moduleNamespace`.
 
-Return a new `WebAssembly.Instance` object setting `[[Instance]]` to `instance`
-and `exports` to `moduleNamespace`.
+Let `instanceObject` be a new `WebAssembly.Instance` object setting
+`[[Instance]]` to `instance` and `exports` to `moduleNamespace`.
+
+If any of the elements of an [Elements section](Modules.md#elements-section)
+refer to an imported function which is not an
+[Exported Function Exotic Object](#exported-function-exotic-objects), throw a
+[`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+
+If, after evaluating the `offset` [initializer expression](Modules.md#initializer-expression)
+of every [Data](Modules.md#data-section) and [Element](Modules.md#elements-section)
+Segment, any of the segments do not fit in their respective Memory or Table, throw a 
+[`RangeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-rangeerror).
+
+Apply all Data and Element segments to their respective Memory or Table in the
+order in which they appear in the module. Segments may overlap and, if they do,
+the final value is the last value written in order. Note: there should be no
+errors possible that would cause this operation to fail partway through. After
+this operation completes, elements of `instance` may are now visible and callable
+through imported Tables, even if `start` fails.
```

Comment:
I am not sure what you mean by _imported Tables_. Could you add a link to the section which defines it?